code/mob/living/critter/sawfly.dm

```diff
@@ -0,0 +1,219 @@
+/*
+->The sawfly, by NightmareChamillian
+This file is the critter itself, and all the custom procs it needs in order to function.
+
+-For the AI, check the critter/AI folder, it should be in sawflyai.dm
+-For the grenade and controller, check code/obj/sawflymisc.dm
+*/
+/mob/living/critter/sawfly
+
+	name = "Sawfly flock code"
+	desc = "A folding antipersonnel drone of syndicate origin. It'd be pretty cute if it wasn't trying to kill people."
+	icon = 'icons/obj/ship.dmi'//remnants of it originally being a drone
+	icon_state = "sawfly"
+	flags = TABLEPASS
+
+	var/beeptext = " "
+	var/dead_state = "sawflydead" //not used- death sprites are handled in the death.dm folder
+	var/angertext = "flies at"
+	var/projectile_type = /datum/projectile/laser/drill/sawfly
+	var/datum/projectile/current_projectile = new/datum/projectile/laser/drill/sawfly
+	var/obj/item/droploot = null // the prox sensors they drop in death are handled in the critterdeath proc
+	var/deathtimer = 0 // for catastrophic failure on death
+	var/isnew = TRUE // for seeing whether or not they will make a new name on redeployment
+	var/sawflynames = list("A", "B", "C", "D", "E", "F", "V", "W", "X", "Y", "Z", "Alpha", "Beta", "Gamma", "Lambda", "Delta")
+	var/beeps = list('sound/machines/sawfly1.ogg','sound/machines/sawfly2.ogg','sound/machines/sawfly3.ogg') // custom noises so they cannot be mistaken for ghostdrones or borgs
+	health = 40
+	var/fliesnearby = 0 //for rolling chance to beep
+	var/friends = list()
+	misstep_chance = 30 //makes them behave more like drones
+
+	hand_count = 1 //stabby hands
+	setup_healths()
+		add_hh_robot(20, 1)
+		add_hh_robot_burn(20, 1)
+	custom_gib_handler = /proc/robogibs
+	blood_id = "oil"
+	use_stamina = FALSE
+	use_stunned_icon = FALSE
+	butcherable = FALSE
+	can_bleed = FALSE
+	canbegrabbed = FALSE
+	can_lie = FALSE
+	can_burn = FALSE
+	pet_text = "cuddles"
+
+
+	New()
+		..()
+		if(isnew)
+			name = "Sawfly [pick(sawflynames)]-[rand(1,999)]"
+		deathtimer = rand(1, 5)
+		death_text = "[src] jutters and falls from the air, whirring to a stop."
+		beeptext = "[pick(list("beeps",  "boops", "bwoops", "bips", "bwips", "bops", "chirps", "whirrs", "pings", "purrs", "thrums"))]"
+		animate_bumble(src) // gotta get the float goin' on
+	//	src.set_a_intent(INTENT_HARM) // incredibly stupid way of ensuring they aren't passable
+		// ai setup
+		src.mob_flags |= HEAVYWEIGHT_AI_MOB
+		src.ai = new /datum/aiHolder/sawfly(src)
+		src.is_npc = TRUE
+		START_TRACKING
+
+	disposing()
+		. = ..()
+		STOP_TRACKING
+
+
+	proc/foldself()
+		var/obj/item/old_grenade/sawfly/reused/N = new /obj/item/old_grenade/sawfly/reused(get_turf(src))
+		// pass our name and health
+		N.name = "Compact [name]"
+		N.tempname = src.name
+		N.temphp = (src.health)
+		qdel(src)
+
+	proc/communalbeep() // distributes the beepchance among the number of sawflies nearby
+		fliesnearby = 1 //that's you, little man! :)
+		for(var/mob/living/critter/sawfly/E in range(get_turf(src), 18))
```

Comment:
Unless you have a hundred sawflys, it will definitely be better to use for_by_tcl here